File: may_challenge/week_2/townJudge.cpp
#include <iostream>
#include <vector>
using namespace std;
class Solution
{
public:
    int findJudge(int N, vector<vector<int>> &trust)
    {
        if (N == 0 || N == 1)
            return N;
        vector<int> arr(N + 1, 0);
        for (auto &t : trust)
            arr[t[1]]++, arr[t[0]]--;
        for (int i = 0; i <= N; i++)
            if (arr[i] == N - 1)
                return i;
        return -1;
    }
};

File: may_challenge/week_2/townJudge.java
class Solution {
    public int findJudge(int N, int[][] trust) {
        int[] arr = new int[N + 1];
        for (int[] t : trust) {
            arr[t[0]]--;
            arr[t[1]]++;
        }
        for (int i = 1; i <= N; i++)
            if (arr[i] == N - 1)
                return i;
        return -1;
    }
}